### PR TITLE
Add missing completions in let variable declaration node context

### LINKED ISSUE #39526: [Bug]: Missing completions in Let variable declaration node context
### Description

Consider the following scenario,

https://user-images.githubusercontent.com/61020198/217173294-f5a7c342-3219-4c8d-b867-25c3b7c625d8.mov

### Steps to Reproduce

```
function testRegExp(string:RegExp name) {
    string: letExpr = let string:<cursor> s = re `${name}` in re `car, van`;
}
```

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Editor

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by mindula**
## Purpose
This PR adds the missing completions in let var decl node context.

Fixes #39526

## Samples

https://user-images.githubusercontent.com/61020198/227847580-7dfb6b44-caa9-41b5-ad16-eb1725bad1cb.mov

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LetVariableDeclarationNodeContext.java [hunk 1]
```diff
@@ -58,38 +58,37 @@ public LetVariableDeclarationNodeContext() {
...
+                completionItems.addAll(this.getCompletionItemList(exprEntries, context));
+                this.sort(context, node, completionItems);
+            } else {
+                /*
+                Covers the following context
+                eg: let va<cursor>
+                    let int a = b, f<cursor>
+                 */
+                completionItems.addAll(this.getTypeDescContextItems(context));
+                completionItems.add(new SnippetCompletionItem(context, Snippet.KW_VAR.get()));
```
Comment (LakshanWeerasinghe) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LetVariableDeclarationNodeContext.java:82]: why are suggesting `var` for the second case? Is it possible to identify cases where only the `var` can be suggested?

Comment (mindula) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LetVariableDeclarationNodeContext.java:82]: Are you suggesting that we handle both these cases separately?

File: language-server/modules/langserver-core/src/test/resources/completion/let_expression_context/source/let_expr_ctx_source17.bal [hunk 2]
```diff
@@ -0,0 +1,3 @@
+public function main() {
+    string:Char ch = string:
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/test/resources/completion/let_expression_context/source/let_expr_ctx_source17.bal:2]: missing let keyword

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LetVariableDeclarationNodeContext.java [hunk 3]
```diff
@@ -58,38 +58,37 @@ public LetVariableDeclarationNodeContext() {
...
+                Covers the following context
+                eg: let var x = <cursor>
+                    let var x = h<cursor>
+                    let var x = mod1:<cursor>
+                */
+                QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) typeDescriptor;
+                List<Symbol> exprEntries = QNameRefCompletionUtil.getExpressionContextEntries(context, qNameRef);
+
+                completionItems.addAll(this.getCompletionItemList(exprEntries, context));
+                this.sort(context, node, completionItems);
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LetVariableDeclarationNodeContext.java:74]: Shall we add an early return here and get rid of the else block?

**Review by malinthar — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/test/resources/completion/let_expression_context/source/let_expr_ctx_source17.bal (hunk 2 above)
Comment (mindula) [reply language-server/modules/langserver-core/src/test/resources/completion/let_expression_context/source/let_expr_ctx_source17.bal]: Fixed in [c9dbc2b](https://github.com/ballerina-platform/ballerina-lang/pull/39994/commits/c9dbc2b76270bb9e5f5f9dbb930c2ce8e9a17b46)

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LetVariableDeclarationNodeContext.java (hunk 3 above)
Comment (mindula) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LetVariableDeclarationNodeContext.java:74]: Done

**Review by malinthar — APPROVED** [review]

**Review by malinthar — APPROVED** [review]

**Review by IMS94 — APPROVED** [review]

**Pull request merged**